java/src/main/java/org/whispersystems/signalservice/loki/api/opengroups/PublicChatAPI.kt

```diff
@@ -287,6 +293,48 @@ class PublicChatAPI(userPublicKey: String, private val userPrivateKey: ByteArray
         }
     }
 
+    public fun updateOpenGroupProfileIfNeeded(channel: Long, server: String, groupId: String, info: LokiPublicChatInfo, groupDatabase: LokiGroupDatabaseProtocol, forceUpdate: Boolean) {
+        //Save user count
+        apiDatabase.setUserCount(channel, server, info.memberCount)
+
+        //Update display name
+        groupDatabase.updateTitle(groupId, info.displayName)
+
+        //Download and update profile picture if needed
+        val oldAvatarURL = apiDatabase.getOpenGroupAvatarURL(channel, server)
+        if (forceUpdate || !Objects.equals(oldAvatarURL, info.profilePictureURL)) {
+            val avatarBytes = downloadOpenGroupAvatar(server, info.profilePictureURL) ?: return
+            groupDatabase.updateAvatar(groupId, avatarBytes)
+            apiDatabase.setOpenGroupAvatarURL(channel, server, info.profilePictureURL)
+        }
+    }
+
+    public fun downloadOpenGroupAvatar(server: String, endpoint: String): ByteArray? {
+        val url = "$server/${endpoint.removePrefix("/")}"
+        Log.v("Loki", "Downloading open group avatar using \"$url\"")
+
+        try {
+            OkHttpClient().newCall(Request.Builder()
```

Comment:
This request has to be onion routed (see the other functions in this file for how to do that)